utils/python/CIME/case.py

```diff
@@ -987,6 +987,12 @@ def get_mpirun_cmd(self, job="case.run"):
             }
 
         executable, args = env_mach_specific.get_mpirun(self, mpi_attribs, job=job)
+        # special case for aprun if using < 1 full node
+        if executable == "aprun":
+            totalpes = self.get_value("TOTALPES")
+            pes_per_node = self.get_value("PES_PER_NODE")
+            if totalpes < pes_per_node:
+                args["tasks_per_node"] = "-N "+str(totalpes)
```

Comment:
The `-n ` argument of qsub also needs a condition to ensure that `$TOTALPES/$PES_PER_NODE` is not 0.